zproject/settings.py

```diff
@@ -226,23 +225,55 @@ def get_secret(key):
 USE_TZ = True
 
 DEPLOY_ROOT = os.path.join(os.path.realpath(os.path.dirname(__file__)), '..')
-TEMPLATE_DIRS = ( os.path.join(DEPLOY_ROOT, 'templates'), )
 LOCALE_PATHS = ( os.path.join(DEPLOY_ROOT, 'locale'), )
 
 # Make redirects work properly behind a reverse proxy
 USE_X_FORWARDED_HOST = True
 
 # List of callables that know how to import templates from various sources.
-TEMPLATE_LOADERS = (
+LOADERS = [
     'django.template.loaders.filesystem.Loader',
     'django.template.loaders.app_directories.Loader',
-    )
+]
 if PRODUCTION:
     # Template caching is a significant performance win in production.
-    TEMPLATE_LOADERS = (
-        ('django.template.loaders.cached.Loader',
-         TEMPLATE_LOADERS),
-        )
+    LOADERS = ['django.template.loaders.cached.Loader'] + LOADERS
+
+TEMPLATES = [
+    {
+        'BACKEND': 'zproject.jinja2.backends.Jinja2',
+        'DIRS': [
+             os.path.join(DEPLOY_ROOT, 'templates'),
+        ],
+        'APP_DIRS': False,
+        'OPTIONS': {
+            'environment': 'zproject.jinja2.environment',
+            'extensions': [
+                'jinja2.ext.i18n',
+                'jinja2.ext.autoescape',
+            ],
+            'context_processors': [
+                'zerver.context_processors.add_settings',
+                'zerver.context_processors.add_metrics',
+            ],
+        },
+    },
+    {
+        'BACKEND': 'django.template.backends.django.DjangoTemplates',
+        'DIRS': [
+             os.path.join(DEPLOY_ROOT, 'templates'),
+        ],
+        'APP_DIRS': False,
+        'OPTIONS': {
+            'debug': DEBUG,
+            'loaders': LOADERS,
+            'context_processors': [
+                'zerver.context_processors.add_settings',
+                'zerver.context_processors.add_metrics',
+            ],
+        },
+    },
+]
```

Comment:
I have separated out directories for Jinja2 and Django templates. Now if Django doesn't find a template in Jinja2 directory it will search in the Django directory and upon finding the template it will process the template using Django template engine.